CHANGELOG.md

```diff
@@ -1,6 +1,10 @@
 Current Version
   -
 
+2.4.0 July 18, 2017
```

Comment:
I'll handle cutting the release (I probably need to make this a more explicit policy)-- please keep your changes under "current version" and then I can make another release commit when we're done with this.